src/object/cli_obj.c

```diff
@@ -882,15 +887,19 @@ obj_shard_tgts_query(struct dc_object *obj, uint32_t map_ver, uint32_t shard,
 	bool			 ec_degrade = false;
 	uint32_t		 ec_deg_tgt = 0, start_shard;
 	bool			 csum_err = false;
+	bool			 tx_uncertain = false;
 	int			 rc;
 
 	shard_tgt->st_ec_tgt = ec_tgt_idx;
 	start_shard = shard - ec_tgt_idx;
 
 	if (obj_auxi->is_ec_obj &&
-	    (obj_auxi->csum_retry ||
+	    (obj_auxi->csum_retry || obj_auxi->tx_uncertain ||
 	     DAOS_FAIL_CHECK(DAOS_OBJ_FORCE_DEGRADE))) {
-		if (obj_auxi->csum_retry) {
+		if (obj_auxi->tx_uncertain) {
+			tx_uncertain = true;
+			obj_auxi->tx_uncertain = 0;
+		} else {
```

Comment:
seems better to be -
} else if (obj_auxi->csum_retry) {

Reply:
right.